rpc/src/optimistically_confirmed_bank_tracker.rs

```diff
@@ -401,14 +401,22 @@ mod tests {
         std::sync::atomic::AtomicU64,
     };
 
-    /// Given a bank get the parent slot chains, this include the parent slot of the oldest parent bank.
+    // Given a bank, get a chain of the parent slots (descending order)
+    // inclusive of the provided bank.
     fn get_parent_chains(bank: &Arc<Bank>) -> Vec<Slot> {
-        let mut parents = bank.parents();
-        let oldest_parent = parents.last().map(|last| last.parent_slot());
-        parents.push(bank.clone());
-        let mut rooted_slots: Vec<_> = parents.iter().map(|bank| bank.slot()).collect();
-        rooted_slots.push(oldest_parent.unwrap_or_else(|| bank.parent_slot()));
-        rooted_slots
+        let parents = bank.clone().parents_inclusive();
```

Comment:
```rust
bank.ancestors.keys()
```

Could be missing a subtle difference, but `Ancestors` seems to already be tracking all of this for us, current bank included. No cloning of arcs necessary.

Reply:
Ahh true, kind of forgot about that field. I think the only subtlety is that we'd no longer get that one extra slot. At runtime, Banks getting cleaned up would mean slots are getting rooted. But, we always keep around a root bank in `BankForks`, and I can't see why the parent of a root would be critical. That being said, this is test code too, so I think using existing functionality is preferred anyways.

I'll make the swap, but debatable whether `ancestors` should even be `pub`